### PR TITLE
- Wire in the guard to make it evident that we have it when making ch…

### PR DESCRIPTION
**Pull request by baldersheim**
…anges that require it.

- Clean up some old members and code not used any more.

@vekterli or @geirst PR

### CODE REVIEW

File: metrics/src/vespa/metrics/metricmanager.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ MetricManager::Timer::getTime() const {
 
 void
 MetricManager::assertMetricLockLocked(const MetricLockGuard& g) const {
-    if ((g.mutex() != &_waiter) || !g.owns_lock()) {
+    if (g.owns(_waiter)) {
```
Comment (geirst) [inline metrics/src/vespa/metrics/metricmanager.cpp:35]: Should this be `!g.owns(_waiter)` instead ?

File: searchcore/src/vespa/searchcore/proton/server/documentdb.h [hunk 2]
```diff
@@ -114,7 +97,6 @@ class DocumentDB : public DocumentDBConfigOwner,
     mutable std::condition_variable _configCV;
     DocumentDBConfig::SP          _activeConfigSnapshot;
     int64_t                       _activeConfigSnapshotGeneration;
-    SerialNum                     _activeConfigSnapshotSerialNum;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/documentdb.h:117]: Why is this removed?

File: searchcore/src/vespa/searchcore/proton/server/proton.cpp [hunk 3]
```diff
@@ -219,7 +232,6 @@ Proton::Proton(const config::ConfigUri & configUri,
       _threadPool(128 * 1024),
       _distributionKey(-1),
       _isInitializing(true),
-      _isReplayDone(false),
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/proton.cpp:222]: Why is this removed?

File: metrics/src/vespa/metrics/updatehook.cpp [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace metrics {
+
+MetricLockGuard::MetricLockGuard(std::mutex & mutex)
+    : _guard(mutex)
+{}
+
+bool
+MetricLockGuard::owns(const std::mutex & mutex) const {
+    return (_guard.mutex() == &mutex) && !_guard.owns_lock();
```
Comment (geirst) [inline metrics/src/vespa/metrics/updatehook.cpp:13]: Should this be `&& _guard.owns_lock();` instead?

File: searchcore/src/vespa/searchcore/proton/server/documentdb.cpp [hunk 5]
```diff
@@ -222,16 +235,14 @@ void DocumentDB::registerReference()
...
-                                 SerialNum serialNum, int64_t generation) {
+void DocumentDB::setActiveConfig(const DocumentDBConfig::SP &config, int64_t generation) {
     lock_guard guard(_configMutex);
     registerReference();
     _activeConfigSnapshot = config;
     assert(generation >= config->getGeneration());
     if (_activeConfigSnapshotGeneration < generation) {
         _activeConfigSnapshotGeneration = generation;
     }
-    _activeConfigSnapshotSerialNum = serialNum;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/documentdb.cpp:234]: Why is this removed?

**Review by geirst — COMMENTED** [review]
I have some questions first.

**Comment by geirst** [comment]
@toregge please take a look as well.

File: searchcore/src/vespa/searchcore/proton/server/documentdb.cpp (hunk 5 above)
Comment (toregge) [reply searchcore/src/vespa/searchcore/proton/server/documentdb.cpp:234]: Some compilers warn about private member variables that appears to be unused. The variable is never read by the C++
code, so the only use might be indirectly (e.g. when inspecting core dumps).

File: metrics/src/vespa/metrics/updatehook.cpp (hunk 4 above)
Comment (toregge) [reply metrics/src/vespa/metrics/updatehook.cpp:13]: Agreed, this looks like an error in the PR.

File: metrics/src/vespa/metrics/metricmanager.cpp (hunk 1 above)
Comment (toregge) [reply metrics/src/vespa/metrics/metricmanager.cpp:35]: Agreed, this looks like an error in the PR.

**Review by toregge — CHANGES_REQUESTED** [review]
`MetricLockGuard::owns()` and `MetricManager::assertMetricLockLocked()` should be fixed.

File: metrics/src/vespa/metrics/metricmanager.cpp (hunk 1 above)
Comment (baldersheim) [reply metrics/src/vespa/metrics/metricmanager.cpp:35]: Good catch. The effect of -1 * -1 = 1.
Fixed

File: searchcore/src/vespa/searchcore/proton/server/proton.cpp (hunk 3 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/proton.cpp:222]: It was not used. It was set, but never used.

File: searchcore/src/vespa/searchcore/proton/server/documentdb.h (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/documentdb.h:117]: It was not used. It was set, but never used. I think it leftovers from when @toregge rewrote how we do TLS replay.

File: metrics/src/vespa/metrics/updatehook.cpp (hunk 4 above)
Comment (baldersheim) [reply metrics/src/vespa/metrics/updatehook.cpp:13]: Yes, 2 bugs cancelling each other causing the tests to still pass.

File: searchcore/src/vespa/searchcore/proton/server/documentdb.cpp (hunk 5 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/documentdb.cpp:234]: I think that having variables around for future possible coredumps is not a good pattern.

**Comment by baldersheim** [comment]
@geirst Fixed, PTAL

**Review by toregge — APPROVED** [review]

**Review by geirst — APPROVED** [review]
👍
